### PR TITLE
is_exception/1 guard

### PR DESCRIPTION
**Pull request by Eiji7**
@josevalim posted [#66](https://elixirforum.com/t/discussion-incorporating-erlang-otp-21-map-guards-in-elixir/14816/66?u=eiji) on forum at `Mar 23, 9:13 AM`:
> (…) is_exception was not implemented but a PR is welcome.

- [x] Added `is_exception/1` guard (based on `Exception.exception?/1` and `Kernel.is_struct/1`)
- [x] Added tests (based on `is_struct/1` tests)
- [x] Updated `CHANGELOG.md`

cc @josevalim

### CODE REVIEW

**Comment by josevalim** [comment]
Looks great, although it is probably best to check for both `__struct__` and `__exception__` fields. Also, let's add `is_exception/2` for completeness!

**Comment by Eiji7** [comment]
@josevalim I have made all changes and as well as updated `CHANGELOG.md`. Is it ok now?

File: CHANGELOG.md [hunk 1]
```diff
@@ -218,7 +218,7 @@ To wrap it all up, `Mix` also includes a new task called `mix app.config`. This
 
 ## Other improvements
 
-Elixir v1.11 adds the `is_struct/2` guard and also includes support for `map.field` syntax in guards.
+Elixir v1.11 adds the `is_struct/2`, `is_exception/1` and `is_exception/2` guards and also includes support for `map.field` syntax in guards.
```
Comment (josevalim) [inline CHANGELOG.md:221]:
```suggestion
Elixir v1.11 adds the `is_struct/2`, `is_exception/1`, and `is_exception/2` guards and also includes support for `map.field` syntax in guards.
```

**Comment by josevalim** [comment]
:green_heart: :blue_heart: :purple_heart: :yellow_heart: :heart:

**Pull request merged**